AntistasiOfficial.Altis/Functions/fn_changeCitySupply.sqf

```diff
@@ -15,38 +9,72 @@ if !(_data isEqualType []) exitWith {diag_log format ["Error in changeCitySupply
 //Execute command
 _supplyLevels = _data select 4;
 
-if(_type == "FOOD") then {_index = 0;};
-if(_type == "WATER") then {_index = 1;};
-if(_type == "FUEL") then {_index = 2;};
+if (_type == "FOOD") then {_index = 0;};
+if (_type == "WATER") then {_index = 1;};
+if (_type == "FUEL") then {_index = 2;};
 
 _currentLevel = _supplyLevels select _index;
 
-if(_delta > 0 && _currentLevel != 'GOOD') then
-{
-	if(_currentLevel == 'LOW') then
-	{
-		_currentLevel = 'GOOD';
-		[0,5,_city] remoteExec ["AS_fnc_changeCitySupport",2];
+diag_log format ["supplylevels food,water,fuel = %1 city = %2", _supplyLevels, _city];
+diag_log format ["type = %1 _currentLevel = %1 city = %2", _type, _currentLevel, _city];
+
+// TODO : dont seem to work properly
+_oldLevel = _currentLevel;
+if (_delta > 0 AND _currentLevel != "GOOD") then {
+    diag_log format ["delta %1", _delta];
+    diag_log format ["_currentLevel %1", _currentLevel];
+    diag_log format ["(_delta > 0 AND _currentLevel != 'GOOD')"];
+
+	if (_currentLevel == "LOW") then {
+        diag_log format ["_currentLevel %1", _currentLevel];
+		_currentLevel = "GOOD";
+        // TODO : need testing
+		[0,5,_city] remoteExec ["AS_fnc_changeCitySupport", 2];
+
+        // TO DO : not working
+        // {
+            // if (_x distance _location < 500) then {[5,_x] call playerScoreAdd};
+        // } forEach (allPlayers - entities "HeadlessClient_F");
+        [5,Slowhand] call playerScoreAdd;
 	};
-	
-	if(_currentLevel == 'CRITICAL') then
+
+	if (_currentLevel == "CRITICAL") then
 	{
-		_currentLevel = 'LOW';
-		[0,15,_city] remoteExec ["AS_fnc_changeCitySupport",2];
+        diag_log format ["_currentLevel %1", _currentLevel];
+		_currentLevel = "LOW";
+        // TODO : need testing
+        [0,10,_city] remoteExec ["AS_fnc_changeCitySupport", 2];
+
+        // TO DO : not working
+        // {
+            // if (_x distance _location < 500) then {[10,_x] call playerScoreAdd};
+        // } forEach (allPlayers - entities "HeadlessClient_F");
+        [10,Slowhand] call playerScoreAdd;
```

Comment:
Only giving score to the commander even if he is AFK at the HQ while some players doing the missions, not really fair, or just for test usage?

Reply:
it's like that in other missions: commander isn't supposed to go, instead he's supposed to stay at HQ and give orders and that's his way to get rewarded

Reply:
basically the mission and his system is here but the impact on gameplay is not tested or implemented. i don't add anything just fixing the existant.

Reply:
Ok but we need a system to reward the acting players too. If the don´t have anything from it the will most likely ignore the missions. Especially if they are on a membership server without being a member

Reply:
of course but first it need to works 😉